core/jobs/batch_jobs/exp_migration_jobs.py

```diff
@@ -76,9 +76,9 @@ def _migrate_exploration(
         """
         try:
             exploration = exp_fetchers.get_exploration_from_model(exp_model)
-            exploration.validate(strict=exp_is_published)
 
             with datastore_services.get_ndb_context():
+                exploration.validate(strict=exp_is_published)
```

Comment:
Maybe add a comment above this line explaining why it needs to be in the "with".